pkg/intermediate/aggregate_test.go

```diff
@@ -694,6 +694,103 @@ func TestGetExpiryFromExpirePriorityQueue(t *testing.T) {
+	assert.Contains(t, recordString, "    flowEndSeconds: 1 \n")
+	assert.Contains(t, recordString, "    flowType: 2 \n")
+	assert.Contains(t, recordString, "    flowEndReason: 2 \n")
+	assert.Contains(t, recordString, "    tcpState: ESTABLISHED \n")
+	assert.Contains(t, recordString, "    ingressNetworkPolicyRuleAction: 0 \n")
+	assert.Contains(t, recordString, "    egressNetworkPolicyRuleAction: 0 \n")
+	assert.Contains(t, recordString, "    ingressNetworkPolicyRulePriority: 50000 \n")
+	assert.Contains(t, recordString, "    packetTotalCount: 500 \n")
+	assert.Contains(t, recordString, "    packetDeltaCount: 0 \n")
+	assert.Contains(t, recordString, "    reversePacketTotalCount: 500 \n")
+	assert.Contains(t, recordString, "    reversePacketDeltaCount: 0 \n")
+}
+
+func TestGetRecords(t *testing.T) {
+	messageChan := make(chan *entities.Message)
+	input := AggregationInput{
+		MessageChan:           messageChan,
+		WorkerNum:             2,
+		CorrelateFields:       fields,
+		ActiveExpiryTimeout:   testActiveExpiry,
+		InactiveExpiryTimeout: testInactiveExpiry,
+	}
+	ap, _ := InitAggregationProcess(input)
+
+	// Add records with IPv4 fields.
+	recordIPv4Src := createDataMsgForSrc(t, false, false, false, false, false).GetSet().GetRecords()[0]
+	recordIPv4Dst := createDataMsgForDst(t, false, false, false, false, false).GetSet().GetRecords()[0]
+	// Add records with IPv6 fields.
+	recordIPv6Src := createDataMsgForSrc(t, true, false, false, false, false).GetSet().GetRecords()[0]
+	recordIPv6Dst := createDataMsgForDst(t, true, false, false, false, false).GetSet().GetRecords()[0]
+	records := []entities.Record{recordIPv4Src, recordIPv4Dst, recordIPv6Src, recordIPv6Dst}
+	for _, record := range records {
+		flowKey, isIPv4, _ := getFlowKeyFromRecord(record)
+		err := ap.addOrUpdateRecordInMap(flowKey, record, isIPv4)
+		assert.NoError(t, err)
+	}
+
+	flowKeyIPv4, _, _ := getFlowKeyFromRecord(recordIPv4Src)
+	partialFlowKeyIPv6 := &FlowKey{
+		SourceAddress: "2001:0:3238:dfe1:63::fefb",
+	}
+	testCases := []struct {
+		name        string
+		flowKey     *FlowKey
+		expectedLen int
+	}{
+		{
+			"Empty flowkey",
+			nil,
+			2,
+		},
+		{
+			"IPv4 flowkey",
+			flowKeyIPv4,
+			1,
+		},
+		{
+			"IPv6 flowkey",
+			partialFlowKeyIPv6,
+			1,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			recordStrings := ap.GetRecords(tc.flowKey)
+			assert.Equalf(t, tc.expectedLen, len(recordStrings), "%s: Number of records string is incorrect, expected %d got %d", tc.name, tc.expectedLen, len(recordStrings))
+			if tc.expectedLen == 1 {
+				assertRecordString(t, recordStrings[0], tc.name == "IPv6 flowkey")
+			} else {
+				if strings.Contains(recordStrings[0], "sourceIPv6Address") {
+					assertRecordString(t, recordStrings[0], true)
+					assertRecordString(t, recordStrings[1], false)
+				} else {
+					assertRecordString(t, recordStrings[0], false)
+					assertRecordString(t, recordStrings[1], true)
+				}
```

Comment:
The prior else block is only for one test case when no flow key is given, right?
What is the use of this if-else block--don't we expect same records every time?